assignments/array-methods.js

```diff
@@ -56,28 +56,77 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+runners.forEach(function(element){fullName.push(element.first_name + ' ' + element.last_name)});
+console.log(fullName);
+
+//or//
+
+let conferenceNames = runners.forEach(function(i){
+  fullName.push(i.first_name + ' ' + i.last_name);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+let allCaps = runners.map(function(element) {return element['first_name'].toUpperCase()});
 console.log(allCaps); 
 
+//or//
+
+let allCaps = runners.map((person) => {
+  return person.first_name.toUpperCase();
+});
+console.log(allCaps);
+
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+
+largeShirts = runners.filter(function(element){
+  return element.shirt_size === 'L';
+});
+console.log(largeShirts);
+
+//or//
+
+let largeShirts = runners.filter((shirts) => {
+  return shirts.shirt_size === 'L';
+});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+
+ticketPriceTotal = runners.reduce((acc, next) => acc + next.donation, 0);
+
+console.log(ticketPriceTotal);
+
+//or//
+
+let ticketPriceTotal = runners.reduce((donationTotal, donationAmount) => {
+  return donationTotal + donationAmount.donation;
+}, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+let email = [];
+runners.forEach(function(element){fullName.push(element.email)});
+console.log(email);
 
 // Problem 2
+let ids = [];
+ids = runners.filter(function(element){
```

Comment:
I don't think this `.filter()` is doing what you think it should. This is looking at every element in the array and checking to see if it has an `id` property, and if it does, then it will return that element. Therefore, since all elements in your array have `id`s, this filter doesn't change anything. Maybe if you did `return element.donation > 100`, then you could get an array of all the runners who donated more than $100. This is actually a practical use-case; maybe you want to put those runners in a unique category to recieve a special "thank you" email, for example.